src/par_iter/slice_mut.rs

```diff
@@ -0,0 +1,67 @@
+use super::*;
+use super::len::ParallelLen;
+use super::state::ParallelIteratorState;
+use std::mem;
+
+pub struct SliceIterMut<'r, T: 'r + Send> {
```

Comment:
I forget why I chose `'r`. Usually I prefer to choose meaningful words, actually, and quite dislike the `'a` "convention" except for small, random lifetimes specific to one fn signature. I guess I was just lazy.